### PR TITLE
[FIXES-#764] bitcompressed_vector's reference type should always be a proxy

### LINKED ISSUE #764: bitcompressed_vector<qualified<dna4, phred42>>::iterator does not model std::InputIterator
One of the issues to get #672 working

Note that `bitcompressed_vector<dna4>::iterator` does model std::InputIterator and `std::vector<qualified<dna4, phred42>>::iterator` does too.

Boils down to this issue:

```cpp
using bit_type = bitcompressed_vector<qualified<dna4, phred42>>;
    std::cout << detail::get_display_name_v<decltype(*std::declval<decltype((std::declval<bit_type>()).begin()) &>())>.string() << std::endl;

```

prints `unsigned char&` which is clearly wrong

### PR DESCRIPTION
**Pull request by h-2**
fixes #764 

added a fix to #679 as well

### CODE REVIEW

**Comment by h-2** [comment]
Why does the test fail on GCC8 and not on the others :sob:

File: include/seqan3/alphabet/detail/alphabet_proxy.hpp [hunk 1]
```diff
@@ -241,6 +241,16 @@ class alphabet_proxy : public alphabet_base<derived_type,
         return assign_rank_to(to_rank(), alphabet_type{});
     }
 
+    //!\brief Implicit conversion to types that the emulated type is convertible to.
+    template <typename other_t>
+    //!\cond
+        requires std::ConvertibleTo<alphabet_type, other_t>
+    //!\endcond
+    constexpr operator other_t() const noexcept
```
Comment (marehr) [inline include/seqan3/alphabet/detail/alphabet_proxy.hpp]:
```suggestion
    constexpr operator other_alphabet_t() const noexcept
```
?

**Comment by h-2** [comment]
@rrahn do want to have a quick peak at this or can I merge it?

**Comment by h-2** [comment]
I have reported a bug for the GCC8 issue: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=90897

This PR already contains a WORKAROUND for it.

**Pull request merged**